### PR TITLE
Made client HTTPErrors more verbose

### PR DESCRIPTION
**Pull request by austin3dickey**
Now errors will print the response content, which sometimes is more verbose than the error message.

### CODE REVIEW

File: tests/unit_tests/test_clients.py [hunk 1]
```diff
@@ -101,3 +103,13 @@ def test_get_comparison_to_baseline(self, conbench_env, z_score_threshold):
...
+
+    @pytest.mark.parametrize("path", ["/error_with_content", "/error_without_content"])
+    def test_client_error_handling(self, conbench_env, path, caplog: LogCaptureFixture):
+        with pytest.raises(requests.HTTPError, match="404"):
+            self.cb.get(path)
+
+        if path == "/error_with_content":
+            assert 'Response content: {"code":' in caplog.text
+        else:
+            assert "Response content: None" in caplog.text
```
Comment (jonkeane) [inline tests/unit_tests/test_clients.py:115]: ❤️

**Review by jonkeane — APPROVED** [review]
This looks good — thanks for keeping the small things up

**Review by alistaire47 — APPROVED** [review]
Nice! I'm feeling a little more urgency to not duplicate this in benchadapt, though I do want to keep its deps near 0. We could have it just depend on benchalert if we don't want to split these off into a dep package for now, but do we foresee benchalert wanting heavier deps in the future? I'm assuming most new regression calculation and such would live on the server, but I'm curious if you have thoughts of things that might live here that require deps.

**Comment by austin3dickey** [comment]
>do we foresee benchalert wanting heavier deps in the future?

Nope. Like you said I think we'd put heavier calculations in the server. This package should mainly be used for munging conbench results into reports, and talking to services like Conbench, GitHub, Slack, etc.

**Comment by alistaire47** [comment]
> > do we foresee benchalert wanting heavier deps in the future?
> 
> Nope. Like you said I think we'd put heavier calculations in the server. This package should mainly be used for munging conbench results into reports, and talking to services like Conbench, GitHub, Slack, etc.

Cool. @jonkeane objections to having benchadapt (and therefore benchrun when it's ready) depend on benchalert? Intuitively strange, but from a maintenance perspective it makes sense

**Comment by jonkeane** [comment]
I would rather we separated the conbench client out cleanly than had either benchalert or benchadapt depend on each other. And in fact, if I had to pick, I would rather benchalert depend on benchadapt compared to the reverse, but again let's just factor the client out to a client package and both of these depend on that client package.

Even "just" the github, slack, etc. dependencies are a something the adapter shouldn't ever need, and the adapter is much more likely to be installed in diverse environments compared to the alerter

**Comment by austin3dickey** [comment]
I made https://github.com/conbench/benchalerts/issues/16 to talk about factoring out the ConbenchClient into its own thing.

**Pull request merged**